semantics/llvm-syntax.k

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2013-2014 K Team. All Rights Reserved.
+/*@
+
+This is the syntax of LLVM.
+
+This syntax requires a input LLVM program without any comments on it.
```

Comment:
Liyi,

You cannot assume implicit what I said in our individual meetings when you write such public documents.  Users of the semantics will not know what we discussed.  I recommend that you talk to Radu about this and see if we can define a syntactic category for such (simple) comments now.  If we can, then we should do it and give empty semantics to comments; for example,

    rule ; Whatever => emptyCommand  [macro]

If we cannot, then you should explain that in the README, and in the testing script, and define an external parser which removes comments from the llvm program and then calls kast on it.  I think this is better than copying all the modified programs somewhere else, for many reasons which we can discuss if you are not convinced.

Grigore


________________________________
From: Liyi Li [notifications@github.com]
Sent: Wednesday, January 14, 2015 9:39 AM
To: kframework/llvm-semantics
Cc: Rosu, Grigore
Subject: Re: [llvm-semantics] Java engine start syntax by defining variables (#5)


In semantics/llvm-syntax.k<https://github.com/kframework/llvm-semantics/pull/5#discussion-diff-22944212>:

> @@ -0,0 +1,29 @@
> +// Copyright (c) 2013-2014 K Team. All Rights Reserved.
> +/*@
> +
> +This is the syntax of LLVM.
> +
> +This syntax requires a input LLVM program without any comments on it.


It is hard to define LLVM comments in the current K framework because the comment separated character is "//" in K; however, the comment separated character is ";" in LLVM. In order to define such comments, we need to define ";" in every possible place it appears in a LLVM program. I think we discussed this issue before and you claimed that we should not worry about this now.

—
Reply to this email directly or view it on GitHub<https://github.com/kframework/llvm-semantics/pull/5/files#r22944212>.